packages/react-router-dom/modules/BrowserRouter.js

```diff
@@ -18,6 +18,12 @@ class BrowserRouter extends React.Component {
   history = createHistory(this.props)
 
   render() {
+    if (this.props.history) {
+      console.error(
+        '`<BrowserRouter history={...}` prop has been ignored. For custom history, ' +
+        'make sure to `import {Router}` and not `import {... as Router}`.')
+    }
+
```

Comment:
when I tried just https://github.com/ReactTraining/react-router/pull/5151/commits/99d971676e7131e7832ffa727c611b161fde906b, the test failed because `console.error` was called 0 times + because there was some spill-over of 2 additional console-errors for unrelated test in `HashRouter`:

```
lerna ERR! execute   A <BrowserRouter>
lerna ERR! execute     × warns when passed a history prop
lerna ERR! execute       Chrome 58.0.3029 (Windows 10 0.0.0)
lerna ERR! execute     Error: Expected 0 to be 1
...
lerna ERR! execute   A <HashRouter>
lerna ERR! execute     × warns when passed a history prop
lerna ERR! execute       Chrome 58.0.3029 (Windows 10 0.0.0)
lerna ERR! execute     Error: Expected 3 to be 1
```

the additional 2 `HashRouter` warnings are following:
```
lerna ERR! execute   A <HashRouter>
lerna ERR! execute     × warns when passed a history prop
lerna ERR! execute       Chrome 58.0.3029 (Windows 10 0.0.0)
lerna ERR! execute     Error: Expected 'Warning: setState(...): Cannot update during an existing state transition (such as within `render` or another component\'s constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.' to match /<HashRouter.*import \{Router}/
```

@pshrmn I have better things to do that trying to debug how to test this :(

I would be happy to add spaces in imports if you think this can be merged without `warning` module and/or someone can figure out the tests :)